vizro-ai/src/vizro_ai/task_pipeline/pipeline.py

```diff
@@ -0,0 +1,60 @@
+"""Task Pipeline."""
+
+from typing import Any, Dict, List, Union
+
+from vizro_ai.chains._llm_models import LLM_MODELS
+
+
+class Pipeline:
+    """A pipeline to manage the flow of tasks in a sequence."""
+
+    def __init__(self, llm: LLM_MODELS):
+        """Initialize the Pipeline.
+
+        Args:
+            llm: The LLM instance to be used by components in the pipeline.
+        """
+        self.llm = llm
+        self.components = []
+        self.components_instances = {}
+
+    def add(self, component_class, input_keys: Union[List, None] = None, output_key: Union[str, None] = None):
+        """Add a component class to the pipeline along with its input and output specifications.
+
+        Args:
+            component_class: The class of the component to be added to the pipeline.
+
+            input_keys: The keys or identifiers for the inputs that this component expects. These should match the
+            output keys of previous components in the pipeline, if applicable.
```

Comment:
Sorry pain in the butt, but can we indent the next line and not use whitespace between arguments